### ISSUE TITLE
Support for http/2 and http/3 

### ISSUE THREAD

**Issue by if-nil** [alecthomas/chroma#773]
**What problem does this feature solve?**

Currently http lexer only recognizes http versions with minor versions, such as `HTTP/1.1` `HTTP/2.0`, but there are tools and websites that prefer `HTTP/2`

For example, the output of the following command `curl --HEAD https://httpbin.org/get -i`
```
HTTP/2 200
date: Sat, 25 Mar 2023 12:33:43 GMT
content-type: application/json
content-length: 257
server: gunicorn/19.9.0
access-control-allow-origin: *
access-control-allow-credentials: true
```
The minor version is also mentioned in [https://http2.github.io/faq/](https://http2.github.io/faq/)

```
Is it HTTP/2.0 or HTTP/2?
The Working Group decided to drop the minor version (“.0”) because it has caused a lot of confusion in HTTP/1.x.

In other words, the HTTP version only indicates wire compatibility, not feature sets or “marketing.”
```

So I hope to add support for `HTTP/2`

Likewise, `HTTP/3` has been officially released and hopefully will also support
